polkit: 123 -> 124

tmuehlbacher opened the pull request:
upstream archived the project on the freedesktop gitlab and migrated to github.

also the new release won't build with Nix anymore, so another patch is necessary.

Changes: https://github.com/polkit-org/polkit/compare/123...124

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtojnar commented on pkgs/development/libraries/polkit/default.nix:
```diff
@@ -58,6 +57,12 @@ stdenv.mkDerivation rec {
       sha256 = "ebbLILncq1hAZTBMsLm+vDGw6j0iQ0crGyhzyLZQgKA=";
     })
+
+    # FIXME: Fixes build for version 124, remove after next release
```
Please add a link to the upstream PR to the comment.

Btw, you should be able to use `PKG_CONFIG_SYSTEMD_SYSUSERS_DIR` environment variable to override it as well.

tmuehlbacher replied:
Thanks for reviewing, will do.

Huh, where is this variable from? Is this just a pkg-config feature I don't know about? I can find `PKG_CONFIG_SYSTEMD_SYSUSERSDIR` used in gnome-initial-setup in nixpkgs but don't know where this is from.

jtojnar replied:
It is [quite recent feature](https://gitlab.freedesktop.org/pkg-config/pkg-config/-/commit/ae0a8b1ad5e40f68fc09ffedac215e038a50f070#124c0becfe68b1ef671f49ed2b9d24779ace126f_341_341). Most online manpages do not seem to know it, even thought it was introduced eight years ago.

tmuehlbacher replied:
Another question about the other patch that is pulled from gitlab.freedesktop.org: Should this be pulled into the repo locally? I don't think that the gitlab instance is going anywhere in the next while but maybe the archived project could be removed eventually? (I don't know any plans from upstream)

jtojnar replied:
Yeah, good idea.

tmuehlbacher replied:
Upstream didn't like the patch, so I instead followed you suggestion and used the switched to the pkg-config environment variables to work around the broken meson option. Think this might ready now. :)

jtojnar commented on pkgs/development/libraries/polkit/default.nix:
```diff
@@ -112,6 +117,7 @@ stdenv.mkDerivation rec {
     "--sysconfdir=/etc"
     "-Dsystemdsystemunitdir=${placeholder "out"}/lib/systemd/system"
+    "-Dsystemdsysusersdir=${placeholder "out"}/lib/sysusers.d"
     "-Dpolkitd_user=polkituser" #TODO? <nixos> config.ids.uids.polkituser
```
Do we not need to drop the following now that polkit ships sysusers.d? https://github.com/NixOS/nixpkgs/blob/e16d9a1782c7e5aaba8de96cbc42d6f47ac5fd72/nixos/modules/security/polkit.nix#L110-L116

tmuehlbacher replied:
I am not entirely sure. There is `systemd.sysusers.enable` but it's marked as experimental and is `false` by default. I don't think there is anything to change, we just ship an additional file in the package, other packages are also doing so. Unless some script scrapes packages for files in those paths? grepping for `sysusers.d` in nixpkgs doesn't yield much or much interesting stuff.

jtojnar replied:
I would expect systemd itself to use those files but maybe that is not enabled in NixOS yet. I have also seen some packages removing the file (e.g. knotdns). No idea if it is necessary.

tmuehlbacher committed:
polkit: move patch from archived upstream to local

since the repo at gitlab.fredesktop.org is archived, it's likely safer
to pull down this patch into nixpkgs directly. in case the archived repo
vanishes in the future.

jtojnar commented on pkgs/development/libraries/polkit/default.nix:
```diff
@@ -108,10 +104,12 @@ stdenv.mkDerivation rec {
     ]))
   ];
 
+  PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR= "${placeholder "out"}/lib/systemd/system";
```
https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/environment-variables-go-to-env.md

tmuehlbacher replied:
Ah! I saw some packages using env but some didn't and I wasn't sure which one was preferred, thanks for the link :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3781

tmuehlbacher committed:
polkit: upstream moved to github

tmuehlbacher committed:
polkit: 123 -> 124

in the new release the meson option `systemdsystemunitdir` is broken by
the new `sysusers.d` code in `meson.build`. we fix this by setting both
of these pkg-config variables through environment variables.

Changes: https://github.com/polkit-org/polkit/compare/123...124

SuperSandro2000 commented on pkgs/development/libraries/polkit/default.nix:
```diff
@@ -38,26 +38,22 @@ let
   patches = [
     # Allow changing base for paths in pkg-config file as before.
     # https://gitlab.freedesktop.org/polkit/polkit/-/merge_requests/100
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

tmuehlbacher replied:
Thanks for reviewing, applied the change.

Also split the update commit into two parts for maybe better provenance for the switch from gitlab to github.

SuperSandro2000 reviewed (approved):
Didn't test it but LGTM

jtojnar reviewed (approved):
Sorry this fell through the cracks, LGTM.
